homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,178 @@
+"""Support for TaHoma Smart Thermostat."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+COMMAND_EXIT_DEROGATION = "exitDerogation"
+COMMAND_REFRESH_STATE = "refreshState"
+COMMAND_SET_DEROGATION = "setDerogation"
+COMMAND_SET_MODE_TEMPERATURE = "setModeTemperature"
+
+CORE_DEROGATED_TARGET_TEMPERATURE_STATE = "core:DerogatedTargetTemperatureState"
+CORE_DEROGATION_ACTIVATION_STATE = "core:DerogationActivationState"
+
+PRESET_FREEZE = "freeze"
+PRESET_NIGHT = "night"
+
+ST_HEATING_MODE_STATE = "somfythermostat:HeatingModeState"
+ST_DEROGATION_HEATING_MODE_STATE = "somfythermostat:DerogationHeatingModeState"
+
+STATE_DEROGATION_FURTHER_NOTICE = "further_notice"
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+STATE_PRESET_AT_HOME = "atHomeMode"
+STATE_PRESET_AWAY = "awayMode"
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+STATE_PRESET_SUDDEN_DROP_MODE = "suddenDropMode"
+
+OVERKIZ_TO_HVAC_MODES = {
+    STATE_DEROGATION_ACTIVE: HVAC_MODE_HEAT,
+    STATE_DEROGATION_INACTIVE: HVAC_MODE_AUTO,
+}
+OVERKIZ_TO_PRESET_MODES = {
+    STATE_PRESET_AT_HOME: PRESET_HOME,
+    STATE_PRESET_AWAY: PRESET_AWAY,
+    STATE_PRESET_FREEZE: PRESET_FREEZE,
+    STATE_PRESET_MANUAL: PRESET_NONE,
+    STATE_PRESET_SLEEPING_MODE: PRESET_NIGHT,
+    STATE_PRESET_SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
+}
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    _attr_hvac_modes = [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+    _attr_preset_modes = [
+        PRESET_NONE,
+        PRESET_FREEZE,
+        PRESET_NIGHT,
+        PRESET_AWAY,
+        PRESET_HOME,
+    ]
+    _attr_min_temp = 15.0
```

Comment:
Do we have a device definition of this device? (do they perhaps save this as attribute or state somewhere?). Or do we need to hardcode it.

Reply:
I don’t think so. @vlebourl can you confirm with your own setup?

Reply:
Here the states returned by the thermostat. I don’t see anything related to a min value.
```
"states": [
          {
            "name": "core:NameState",
            "type": 3,
            "value": "** **"
          },
          {
            "name": "somfythermostat:HeatingModeState",
            "type": 3,
            "value": "manualMode"
          },
          {
            "name": "core:TargetTemperatureState",
            "type": 2,
            "value": 15.0
          },
          {
            "name": "somfythermostat:AtHomeTargetTemperatureState",
            "type": 2,
            "value": 20.5
          },
          {
            "name": "somfythermostat:SleepingModeTargetTemperatureState",
            "type": 2,
            "value": 19.0
          },
          {
            "name": "somfythermostat:AwayModeTargetTemperatureState",
            "type": 2,
            "value": 16.0
          },
          {
            "name": "somfythermostat:GeofencingModeTargetTemperatureState",
            "type": 2,
            "value": 20.5
          },
          {
            "name": "somfythermostat:FreezeModeTargetTemperatureState",
            "type": 2,
            "value": 3.14
          },
          {
            "name": "somfythermostat:SuddenDropModeTargetTemperatureState",
            "type": 2,
            "value": 16.0
          },
          {
            "name": "core:BatteryLevelState",
            "type": 2,
            "value": 39.0
          },
          {
            "name": "core:AvailabilityState",
            "type": 3,
            "value": "available"
          },
          {
            "name": "somfythermostat:DerogationTypeState",
            "type": 3,
            "value": "further_notice"
          },
          {
            "name": "core:DerogationStartDateTimeState",
            "type": 5,
            "value": 1648612871000
          },
          {
            "name": "core:DerogatedTargetTemperatureState",
            "type": 2,
            "value": 15.0
          },
          {
            "name": "somfythermostat:DerogationHeatingModeState",
            "type": 3,
            "value": "manualMode"
          },
          {
            "name": "core:DerogationActivationState",
            "type": 3,
            "value": "active"
          }
        ],
```

Reply:
I can confirm the 15°C isn't shown in any attribute as a min value but is indeed the min possible value there. 
<img width="486" alt="image" src="https://user-images.githubusercontent.com/5701372/161545079-5915db3c-70dd-4af1-b17c-80788764b319.png">
Setting the temperature lower than 15°C is the same as setting `freezeMode`.